earthfile2llb/with_docker_run_local_reg.go

```diff
@@ -42,7 +45,8 @@ func (w *withDockerRunLocalReg) Run(ctx context.Context, args []string, opt With
 	}
 
 	defer func() {
-		cmd.SetEndError(retErr)
+		message := solvermon.FormatError("WITH DOCKER RUN", retErr.Error())
+		cmd.SetEnd(time.Now(), logstream.RunStatus_RUN_STATUS_FAILURE, message)
```

Comment:
this feels wrong, this will be called even when `retErr == nil`. Won't this always mark a run as failed?

Reply:
Yep. Fixed